pkg/kubelet/kubelet.go

```diff
@@ -1766,10 +1772,16 @@ func (kl *Kubelet) syncTerminatingPod(ctx context.Context, pod *v1.Pod, podStatu
 	kl.statusManager.SetPodStatus(pod, apiPodStatus)
 
 	if gracePeriod != nil {
+		if *gracePeriod <= 1 {
+			// If we plan to terminate quickly, stop probes immediately, otherwise we will  wait until the pod is completely done
+			kl.probeManager.RemovePod(pod)
```

Comment:
should we try to optimize this?

Reply:
If they are short grace period pods, then I think this is ok.

Reply:
Technically this is to allow a user visible difference between “stop as fast as possible”, and every other kind of stop.  We do that by clamping the grace period to this “less than or equal to one second” range, and since probe periods are quantized by seconds, 1 or below is effectively the same.  And, when someone wants to shut a bunch of pods down as fast as possible, that usually involves freeing up resources or other urgent behavior.

This should also reduce the frequency of noise on shutdown due to probes racing with termination, so it has a net benefit.

I think as long as the comment includes a bit more context about why this is ok, it should be fine.

Reply:
I updated the comment in the latest push. Thanks!

Reply:
@smarterclayton I've found the related issue https://github.com/kubernetes/kubernetes/pull/102025 , is 0 still having a special meaning? and we keeping its behavior with this change?

Reply:
main concern is this last comment https://github.com/kubernetes/kubernetes/pull/102025#issuecomment-896327232

Reply:
I've tried to follow all the graceperiods paths but I'm not able to fully comprehend it , there is this function that seems to always set it to 1
https://github.com/kubernetes/kubernetes/blob/3eadd1a9ead7a009a9abfbd603a5efd0560473cc/pkg/kubelet/pod_workers.go#L641-L671

should we make the optimization at least in a different commit, so we don't mix it with the bugfix?

Reply:
+1. As @aojea mentioned, perhaps we should include this optimization in a followup PR (or separate commit) entirely. 

https://github.com/kubernetes/kubernetes/pull/102025 did not actually merge so we don't consider "1" as a special grace period and since we plan to backport this PR to 1.23 and 1.22, it probably makes sense to avoid adding this optimization since #102025 was not merged. What do you think @rphillips ? Or is there some unintended side affects without this optimization?